prov/gni/src/gnix_rma.c

```diff
@@ -740,6 +795,152 @@ int _gnix_rma_post_rdma_chain_req(void *data)
 	return FI_SUCCESS;
 }
 
+static void __gnix_rma_more_fill_pd(struct gnix_fab_req *req,
+				    struct gnix_tx_descriptor *txd)
+{
+	gni_ct_put_post_descriptor_t *more_put;
+	gni_ct_get_post_descriptor_t *more_get;
+	gni_mem_handle_t mdh;
+	struct gnix_fab_req *more_req;
+	struct slist_entry *current;
+	int entries = 0, idx = 0;
+
+	current = req->rma.sle.next; /* First sub descriptor slist entry */
```

Comment:
After our discussion, and discussing with @ztiffany 
I've added a local macro to be used that makes this code less ugly.